doyensec/detectors/weblogic_cve_2024_21181/README.md

```diff
@@ -0,0 +1,70 @@
+# Weblogic T3/IIOP Deserialization (CVE-2024-21181)
+
+## Description
+
+CVE-2024-21181 is a critical vulnerability in Oracle WebLogic Server. The vulnerability lies in the way it
+handles T3/IIOP requests. When performing the lookup for a Reference object the unsafe deserialization is triggered.
+
+### Detector's implementation
+
+This detector only exploits the deserialization vulnerability to perform a simple DNS callback. 
+Even though this doesn't leak any sensitive data, it hints that a more complex gadget chain is possible.
+Implementing a gadget-chain that leverages the deserialization vulnerability to achieve a complete RCE
+it's outside the scope of the scanner.
+
+**The detector needs the Tsunami Callback Server with the DNS mode enabled.**
+
+## Affected Versions
+
+-  WebLogic Server 12 <= v12.2.1.4.0
+-  WebLogic Server 14 <= v14.1.1.0.0
+
+## Build the plugin
+
+### Oracle Library
+
+The plugin needs the `wlclient.jar` library from Oracle WebLogic to communicate with the Oracle WebLogic server, but since it's proprietary software we can't include it in the repo.
```

Comment:
Is the jar still needed with the current implementation?

Reply:
Hi @maoning, yes it is still required for the detection to work. Without it, the plugin will still compile (as to not disrupt the setup process), but it will exit early with a warning during the detection phase asking the user to recompile it with the Oracle library. Note that the warning is only printed AFTER fingerprinting, if a target is actually confirmed to be Weblogic, so it won't appear during scans on unrelated software.